src/librustdoc/clean/mod.rs

```diff
@@ -1833,6 +1833,10 @@ pub(crate) fn clean_middle_ty<'tcx>(
 
         ty::Alias(ty::Projection, ref data) => clean_projection(bound_ty.rebind(*data), cx, def_id),
 
+        // FIXME(fmease): Clean inherent projections properly. This requires making the trait ref in
+        // `QPathData` optional or alternatively adding a new `clean::Type` variant.
+        ty::Alias(ty::Inherent, _data) => Type::Infer,
```

Comment:
Not rustdoc person but making the trait ref optional does seem doable...

Reply:
I've done a couple of rustdoc PRs touching `clean/mod.rs` and yes, definitely doable. Will do